src/connector/src/parser/unified/json.rs

```diff
@@ -393,26 +406,67 @@ impl JsonParseOptions {
                     .into()
             }
             (DataType::Decimal, ValueType::I64 | ValueType::U64) => {
-                Decimal::from(value.try_as_i64().map_err(|_| create_error())?).into()
+                let i64_val = value.try_as_i64().map_err(|_| create_error())?;
```

Comment:
A `value` of `ValueType::U64` will fail on `try_as_i64`. Is this by design?

Reply:
You are right it would be an error and be filled with null.

> Is this by design?

It is by poor design. Decimals in JSON shall be encoded as string rather than number. Support of i64/u64 in json are non-standard extensions and we do not guarantee its end-to-end quality. This is a fundamental problem of json itself. If the user insist on using json number rather than json string, they are at their own risk to have data loss when it is larger than MAX_SAFE_INTEGER (2^53-1), according to RFC 8259

To summarize, a fix is good to have but not required.